cmd/kubeadm/app/cmd/upgrade/plan.go

```diff
@@ -50,25 +55,199 @@ func newCmdPlan(apf *applyPlanFlags) *cobra.Command {
 		applyPlanFlags: apf,
 	}
 
+	outputFlags := newUpgradePlanPrintFlags(output.TextOutput)
+
 	cmd := &cobra.Command{
 		Use:   "plan [version] [flags]",
 		Short: "Check which versions are available to upgrade to and validate whether your current cluster is upgradeable. To skip the internet check, pass in the optional [version] parameter",
 		RunE: func(_ *cobra.Command, args []string) error {
-			return runPlan(flags, args)
+			printer, err := outputFlags.ToPrinter()
+			if err != nil {
+				return errors.Wrap(err, "could not construct output printer")
+			}
+
+			return runPlan(flags, args, printer)
 		},
 	}
 
+	outputFlags.AddFlags(cmd)
+
 	// Register the common flags for apply and plan
 	addApplyPlanFlags(cmd.Flags(), flags.applyPlanFlags)
 	return cmd
 }
 
+// newComponentUpgradePlan helper creates outputapiv1alpha2.ComponentUpgradePlan object
+func newComponentUpgradePlan(name, currentVersion, newVersion string) outputapiv1alpha2.ComponentUpgradePlan {
+	return outputapiv1alpha2.ComponentUpgradePlan{
+		Name:           name,
+		CurrentVersion: currentVersion,
+		NewVersion:     newVersion,
+	}
+}
+
+// upgradePlanPrintFlags defines printer flag structure for the
+// upgrade plan kubeadm command and provides a method
+// of retrieving a known printer based on flag values provided.
+type upgradePlanPrintFlags struct {
+	// JSONYamlPrintFlags provides default flags necessary for json/yaml printing.
```

Comment:
```suggestion
	// JSONYamlPrintFlags provides default flags necessary for json/yaml printing
```

let's keep comments without trailing dot, for some consistency...